.github/workflows/ci.yaml

```
@@ -26,6 +26,14 @@ on:
     - cron: '0 3 * * *'  # Every day at 3 AM
   workflow_dispatch:
 
+# Cancels any in-progress runs within the same group identified by workflow name and GH reference (branch or tag) 
+# For example it would:
+# - terminate previous PR CI execution after pushing more changes to the same PR branch
+# - terminate previous on-push CI run after merging new PR to main 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

@WojciechMazur That would be a good compromise 😄 
```suggestion
  cancel-in-progress: ${{ github.event_name == 'push }}
```

---

Looks like a typo in the suggestion: missing `'` after `'push`

---

```suggestion
  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```
that should be the other way around no? We don't want to cancel in progress if it's main. That's what we are already doing in metals and it's working ok

---

`base_ref` is available only in pull requests. So it would cancel stale jobs triggered on pull requests to main branch. 
Tomasz's version would allow cancelling in all branches other than main, either on push or pull requests. I think that is closer to our goals.

> The base_ref or target branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.